plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/standard/StandardDeployJob.java

```diff
@@ -0,0 +1,46 @@
+package com.google.cloud.tools.eclipse.appengine.deploy.standard;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+
+import com.google.cloud.tools.eclipse.appengine.deploy.AppEngineDeployJob;
+
+public class StandardDeployJob implements AppEngineDeployJob {
+  
+  private ProjectToStagingExporter projectToStagingExporter;
+  private StagingDirectoryProvider stagingDirectoryProvider;
+  private IProject project;
+
+  public StandardDeployJob(ProjectToStagingExporter exporter, 
+                           StagingDirectoryProvider stagingDirectoryProvider,
+                           IProject project) {
+    projectToStagingExporter = exporter;
+    this.stagingDirectoryProvider = stagingDirectoryProvider;
+    this.project = project;
+  }
+
+  @Override
+  public void run(IProgressMonitor monitor) throws CoreException {
+    String stageDir = stagingDirectoryProvider.getStagingDirectory();
+    if (stageDir == null) {
+      // TODO move /com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/Activator.java
+      // to a bundle that can be a common dependency for all bundles, and use PLUGIN_ID instead of string literal
+      throw new CoreException(new Status(IStatus.ERROR, "pluginid", "Staging directory cannot be null"));
+    }
+    writeProjectToStageDir(project, stageDir);
+    verifyProjectInStageDir(stageDir);
+    //TODO run stage and deploy operations
+  }
+  
+  void writeProjectToStageDir(IProject project, String stageDir) throws CoreException {
+    projectToStagingExporter.writeProjectToStageDir(project, stageDir);
+  }
+
+  void verifyProjectInStageDir(String stageDir) {
```

Comment:
Because the staging directory might be created on a different file system as the Eclipse workspace and then the files are not guaranteed to be copied over during publish.